src/Compilers/CSharp/Test/Symbol/Symbols/FunctionPointerTypeSymbolTests.cs

```diff
@@ -692,5 +698,92 @@ .maxstack  8
+                    // (4,21): warning CS0169: The field 'C._field' is never used
+                    //     delegate*<void> _field;
+                    Diagnostic(ErrorCode.WRN_UnreferencedField, "_field").WithArguments("C._field").WithLocation(4, 21),
+                    // (5,5): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                    //     delegate*<void> Property { get; set; }
+                    Diagnostic(ErrorCode.ERR_UnsafeNeeded, "delegate*").WithLocation(5, 5),
+                    // (6,5): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                    //     delegate*<void> M(delegate*<void> param)
+                    Diagnostic(ErrorCode.ERR_UnsafeNeeded, "delegate*").WithLocation(6, 5),
+                    // (6,23): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                    //     delegate*<void> M(delegate*<void> param)
+                    Diagnostic(ErrorCode.ERR_UnsafeNeeded, "delegate*").WithLocation(6, 23),
+                    // (8,9): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                    //         delegate*<void> local1;
+                    Diagnostic(ErrorCode.ERR_UnsafeNeeded, "delegate*").WithLocation(8, 9),
+                    // (8,25): warning CS0168: The variable 'local1' is declared but never used
+                    //         delegate*<void> local1;
+                    Diagnostic(ErrorCode.WRN_UnreferencedVar, "local1").WithArguments("local1").WithLocation(8, 25),
+                    // (9,13): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                    //         /**/delegate/**/*/**/<void> local2;
+                    Diagnostic(ErrorCode.ERR_UnsafeNeeded, "delegate/**/*").WithLocation(9, 13),
+                    // (9,37): warning CS0168: The variable 'local1' is declared but never used
+                    //         /**/delegate/**/*/**/<void> local2;
+                    Diagnostic(ErrorCode.WRN_UnreferencedVar, "local2").WithArguments("local2").WithLocation(9, 37));
+        }
+
+        [Fact]
+        public void MisdeclaredArraysWithLocalDeclarationsAreHandled()
+        {
+            var comp = CreateFunctionPointerCompilation(@"
+class C
+{
+    unsafe void M()
+    {
+        int a = 1;
+        delegate*<int[a]> local;
+    }
+}");
+
+            comp.VerifyDiagnostics(
+                    // (6,13): warning CS0219: The variable 'a' is assigned but its value is never used
+                    //         int a = 1;
+                    Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "a").WithArguments("a").WithLocation(6, 13),
+                    // (7,22): error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)
```

Comment:
>try initializing with a 'new' expression [](start = 103, length = 40)

It doesn't look like this portion is applicable to the situation, but it is probably preexisting condition.

Reply:
Do you think this is worth adding a new error for it? Personally, I can't see this particular scenario happening very often.

---
In reply to: [376485842](https://github.com/dotnet/roslyn/pull/41455#discussion_r376485842) [](ancestors = 376485842)

Reply:
> Do you think this is worth adding a new error for it?

It might be worth opening an issue, I assume there are scenarios not related to this feature that are also affected. Like within type arguments, etc.

---
In reply to: [376608344](https://github.com/dotnet/roslyn/pull/41455#discussion_r376608344) [](ancestors = 376608344,376485842)

Reply:
[https://github.com/dotnet/roslyn/issues/41494](https://github.com/dotnet/roslyn/issues/41494)

---
In reply to: [376609526](https://github.com/dotnet/roslyn/pull/41455#discussion_r376609526) [](ancestors = 376609526,376608344,376485842)